std/regex/internal/tests.d

```diff
@@ -1082,3 +1082,21 @@ alias Sequence(int B, int E) = staticIota!(B, E);
     assert(equal!equal(s.matchAll(ctr), outcomes));
     assert(equal!equal(s.bmatch(r), outcomes));
 }
+
+
+// bugzilla 17668
+@safe unittest
+{
+    import std.algorithm.searching;
+    bool thrown = false;
+    try
+    {
+        regex(q"<[^]>");
+    }
+    catch (RegexException e)
+    {
+        assert(e.msg.canFind("no operand for '^'"));
+        thrown = true;
+    }
+    assert(thrown);
+}
```

Comment:
Done, for some reason it seemed to not work the first time I tried i.e. exception escaped.